src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1297,7 +1273,13 @@ void ImportStoreIndirect(TypeDesc type)
             var address = Pop();
 
             Check(!address.IsReadOnly, VerifierError.ReadOnlyIllegalWrite);
-            CheckIsAddressCompatibleWithType(address, type);
+
+            CheckIsByRef(address);
+            if (type != null)
+                CheckIsAssignablePointer(type, address.Type);
```

Comment:
I deleted the `CheckIsAssignablePointer` method and replaced the two occurrences with `CheckIsAssignable`.